src/sqlite/duckdbvfs.cpp

```diff
@@ -0,0 +1,696 @@
+    duckdbFileControl,              /* xFileControl */
+    duckdbSectorSize,               /* xSectorSize */
+    duckdbDeviceCharacteristics     /* xDeviceCharacteristics */
+  };
+D_ASSERT(duckdb::getdb());
+       auto &fs = duckdb::getdb()->GetFileSystem();
+
+  DemoFile *p = (DemoFile*)pFile; /* Populate this structure */
+  int oflags = 0;                 /* flags to pass to open() call */
+  char *aBuf = 0;
+
+  if( zName==0 ){
+    return SQLITE_IOERR;
+  }
+
+  if( flags&SQLITE_OPEN_MAIN_JOURNAL ){
+    aBuf = (char *)sqlite3_malloc(SQLITE_DEMOVFS_BUFFERSZ);
+    if( !aBuf ){
+      return SQLITE_NOMEM;
+    }
+  }
+
+  if( flags&SQLITE_OPEN_CREATE )    oflags |= duckdb::FileFlags::FILE_FLAGS_FILE_CREATE;
+  if( flags&SQLITE_OPEN_READONLY )  oflags |= duckdb::FileFlags::FILE_FLAGS_READ;
+  if( flags&SQLITE_OPEN_READWRITE ) oflags |= duckdb::FileFlags::FILE_FLAGS_READ | duckdb::FileFlags::FILE_FLAGS_WRITE;
+  memset(p, 0, sizeof(DemoFile));
+              auto file = fs.OpenFile(zName, oflags);
+             p->fd= file.release();
+
+  if( !p->fd ){
+    sqlite3_free(aBuf);
+    return SQLITE_CANTOPEN;
+  }
+  p->aBuffer = aBuf;
+
+  if( pOutFlags ){
+    *pOutFlags = flags;
+  }
+  p->base.pMethods = &duckdbio;
+  return SQLITE_OK;
+}
+
+/*
+** Delete the file identified by argument zPath. If the dirSync parameter
+** is non-zero, then ensure the file-system modification to delete the
+** file has been synced to disk before returning.
+*/
+static int duckdbDelete(sqlite3_vfs *pVfs, const char *zPath, int dirSync){
+  //FIXME: This might need to be properly implemented
+  return SQLITE_OK;
+
+//  int rc;                         /* Return code */
+
+//  rc = unlink(zPath);
+//  if( rc!=0 && errno==ENOENT ) return SQLITE_OK;
+
+//  if( rc==0 && dirSync ){
+//    int dfd;                      /* File descriptor open on directory */
+//    char *zSlash;
+//    char zDir[MAXPATHNAME+1];     /* Name of directory containing file zPath */
+
+    /* Figure out the directory name from the path of the file deleted. */
+//    sqlite3_snprintf(MAXPATHNAME, zDir, "%s", zPath);
+//    zDir[MAXPATHNAME] = '\0';
+//    zSlash = strrchr(zDir,'/');
+//    if( zSlash ){
+//      /* Open a file-descriptor on the directory. Sync. Close. */
+//      zSlash[0] = 0;
+//      dfd = open(zDir, O_RDONLY, 0);
+//      if( dfd<0 ){
+//        rc = -1;
+//      }else{
+//        rc = fsync(dfd);
+//        close(dfd);
+//      }
+//    }
+//  }
+//  return (rc==0 ? SQLITE_OK : SQLITE_IOERR_DELETE);
+}
+
+#ifndef F_OK
+# define F_OK 0
+#endif
+#ifndef R_OK
+# define R_OK 4
+#endif
+#ifndef W_OK
+# define W_OK 2
+#endif
+
+/*
+** Query the file-system to see if the named file exists, is readable or
+** is both readable and writable.
+*/
+static int duckdbAccess(
+  sqlite3_vfs *pVfs, 
+  const char *zPath, 
+  int flags, 
+  int *pResOut
+){
+  //FIXME: This might need to be properly implemented
+  *pResOut = false;
```

Comment:
The problem with removing this is that we break file locking of SQLite - which means the SQLite scanner is no longer safe to use with other SQLite instances